tools/registry.c

```diff
@@ -18,6 +18,9 @@
 #include "data.h"
 #include "sym.h"
 
+// Helper macro to actually do return checks
+#define NON_ZERO_RETURN( A ) { int result = A; if ( result != 0 ) { printf( "Error in %s, non-zero return expected", #A ); exit(result); } }
```

Comment:
Fair, how about `EXIT_ON_NONZERO` or something to that effect?
I think it is important to specify that non-zero returns are the failure mode, as failure codes can often have different ranges ( -1, non-zero, true/false in which case might be language specific about how bools are implemented)

Reply:
Ideally, we wouldn't depend on a specific, hard-coded integer value to indicate success. Instead, in my opinion it would be better to define a macro or constant to identify the "success" return code, and in this case, calling the macro `EXIT_ON_FAILURE` would probably be reasonable, since it could be defined as:
```
#define EXIT_ON_FAILURE( A ) { int result = A; if ( result != REGISTRY_SUCCESS ) { printf( "Error in %s, return code = %i\n", #A , result); exit(result); } }
```
However, I think `EXIT_ON_NONZERO` is acceptable given that the macro explicitly compares the returned value with `0`.

Reply:
Also, should the message say "zero return expected" rather than "non-zero return expected"?